VmAgent.Core/Interfaces/SystemOperations.cs

```diff
@@ -261,5 +283,45 @@ public void SetUnixFilePermissions(string filePath, int permissions)
             var unixFileInfo = new UnixFileInfo(filePath);
             unixFileInfo.FileAccessPermissions = (FileAccessPermissions)permissions;
         }
+
+        private void SetUnixOwner(string filePath, string owner)
+        {
+            var passwd = Syscall.getpwnam(owner);
+            if(passwd == null)
+            {
+                throw new ArgumentException($"The specified user {owner} could not be retrieved");
+            }
+            var result = Syscall.chown(filePath, passwd.pw_uid, passwd.pw_gid);
+            if(result != 0)
+            {
+                throw new ArgumentException($"Failed to give ownership of {filePath} to {owner}");
+            }
+        }
+
+        public void SetUnixOwnerIfNeeded(string path)
+        {
+            if (_vmConfiguration != null && _vmConfiguration.RunContainersInUsermode)
+            {
+                FileAttributes attr = File.GetAttributes(path);
+                if ((attr & FileAttributes.Directory) == FileAttributes.Directory)
+                {
+                    DirectoryInfo dirInfo = new DirectoryInfo(path);
+                    IEnumerable<FileInfo> files = dirInfo.EnumerateFiles("*", SearchOption.AllDirectories);
```

Comment:
Instead of calling EnumerateFiles and EnumerateDirectories separately, you can call EnumerateFileSystemInfos and that will return both files and directories